python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -90,11 +94,106 @@ def is_managed_object(self, obj_id: str) -> bool:
         """
         return obj_id in self.managed_gpu_object_metadata
 
+    def add_pending_gpu_object(
+        self, obj_ref: ObjectRef, gpu_object_meta: GPUObjectMeta
+    ):
+        """
+        Add the pending receive GPU object to the GPU object manager.
+
+        Args:
+            obj_ref: The ObjectRef of the GPU object.
+            gpu_object_meta: The GPU object metadata.
+        """
+        obj_id = obj_ref.hex()
+        self.pending_gpu_object_metadata[obj_id] = gpu_object_meta
+
+    def is_pending_gpu_object(self, obj_ref: ObjectRef) -> bool:
+        """
+        Check if the GPU object is pending.
+
+        Args:
+            obj_ref: The ObjectRef of the GPU object.
+
+        Returns:
+            True if the GPU object is pending, False otherwise.
+        """
+        obj_id = obj_ref.hex()
+        return obj_id in self.pending_gpu_object_metadata
+
+    def receive_pending_gpu_object(self, obj_ref: ObjectRef, use_object_store: bool):
```

Comment:
The logic in this function seems like it's duplicating some logic elsewhere. I think it can be simplified by addint a tensor_transport arg to `fetch_object` and calling __ray_recv__ there for non-object store backends?